lcls_tools/common/devices/yaml/generate.py

```diff
@@ -43,20 +46,35 @@ def __init__(
 
     def _filter_elements_by_fields(self, required_fields: List[str]) -> Dict[str, Any]:
         csv_reader = None
-        with open(self.csv_location, "r") as file:
+        with (
+            open(self.csv_location, "r") as file_csv,
+            open(self.filter_location, "r") as file_filter,
+        ):
             # convert csv file into dictionary for filtering
-            csv_reader = csv.DictReader(f=file)
+            csv_reader = csv.DictReader(f=file_csv)
+            filter_dict = yaml.safe_load(file_filter)
+
+            def _is_filtered_row(element: dict):
+                released = True
+                for key in set(filter_dict.keys()) & set(element.keys()):
+                    value = element[key]
+                    for prefix in filter_dict[key]:
+                        released &= not value.startswith(prefix)
+                return released
+
+            elements = list(filter(_is_filtered_row, csv_reader))
 
             # make the elements from csv stripped out with only information we need
             def _is_required_field(pair: tuple):
-                key, _ = pair
+                key, value = pair
                 return key in required_fields
 
             # only store the required fields from lcls_elements, there are lots more!
             elements = [
```

Comment:
I'd rather rename 'elements' either on line 65 or here. I think technically this could work fine, but makes me nervous.